andela/haven-ah-backend
# #162234039 Create endpoint to set featured author

#### What does this PR do?
Create endpoint to set author of the week. **PUT** `/api/v1/admin/authors/:username`.
Create endpoint to get the author of the week. **GET** `/api/v1/authors/featured`.

#### Description of Task to be completed
- write migration to update users table to include `isFeaturedAuthor` column.
- write function to update `isFeaturedAuthor` column of the table.
- test functions and routes with mocha and chai.
- add swagger documentation for endpoint

#### How should this be manually tested?
- on the terminal run `git clone https://github.com/andela/haven-ah-backend.git`.
- run `git checkout ft-get-author_of_the_week-162234039`.
- ensure that PostgreSQL is running on your local machine.
- set up databases (test and development) following the .env.sample file.
- run `npm install` to install the npm packages.
- run `npm test` to run the unit and integration tests.
To test on postman

run `npm run start-dev` and open up postman.

**Note:** Authentication is required to set a featured author. Only users with admin roles can perform this operation.

#### Screenshots where applicable
<img width="1091" alt="screen shot 2018-11-28 at 11 34 34 pm" src="https://user-images.githubusercontent.com/29674347/49186934-3b2fc400-f366-11e8-83b1-951e5523d356.png">

<img width="1095" alt="screen shot 2018-11-28 at 11 28 01 pm" src="https://user-images.githubusercontent.com/29674347/49186828-f60b9200-f365-11e8-8609-e5ee07c36dd5.png">

<img width="1101" alt="screen shot 2018-11-28 at 11 28 55 pm" src="https://user-images.githubusercontent.com/29674347/49186677-93b29180-f365-11e8-80a1-57c77b22b3bb.png">

#### Relevant Pivotal Tracker stories
#162234039
